components/switch/switch-mixin.js

```diff
@@ -197,16 +208,20 @@ export const SwitchMixin = superclass => class extends FocusMixin(RtlMixin(Focus
 				@keyup="${this._handleKeyUp}"
 				role="switch"
 				tabindex="${ifDefined(tabindex)}">
-				<div class="d2l-switch-inner">
+				<div class="d2l-switch-inner ${classMap(innerSwitchClasses)}">
 					<div class="d2l-switch-toggle"><div></div></div>
 					<div class="d2l-switch-icon-on">${this.onIcon}</div>
 					<div class="d2l-switch-icon-off">${this.offIcon}</div>
 				</div>
 			</div>
-			${textPosition === 'end' ? text : ''}
+			${textPosition === 'end' || textPosition === 'hidden' ? switchLabel : ''}
 		`;
 	}
 
+	get _labelContent() {
+		return html`<span @click='${this._handleClick}' @mouseenter='${this._handleSwitchHover}' @mouseleave='${this._handleSwitchHoverLeave}'>${this.text}</span>`;
```

Comment:
nit: if you're fixing anything else, this line is long and could be split over multiple